extensions/interactions/MusicNotesInput/directives/oppia-interactive-music-notes-input.directive.ts

```diff
@@ -780,18 +780,23 @@ angular.module('oppia').directive('oppiaInteractiveMusicNotesInput', [
         // is treated like a chord and all its values are played back
         // simultaneously.
         var playSequence = function(midiSequence) {
-          var notes = [];
-          for (var i = 0; i < midiSequence.length; i++) {
-            for (var j = 0; j < midiSequence[i].length; j++) {
-              notes.push({
-                midiValue: midiSequence[i][j],
-                duration: 1.0,
-                start: getNoteStart(i)
-              });
+          if (window.AudioContext || window.Audio) {
+            var notes = [];
+            for (var i = 0; i < midiSequence.length; i++) {
+              for (var j = 0; j < midiSequence[i].length; j++) {
+                notes.push({
+                  midiValue: midiSequence[i][j],
+                  duration: 1.0,
+                  start: getNoteStart(i)
+                });
+              }
             }
-          }
 
-          MusicPhrasePlayerService.playMusicPhrase(notes);
+            MusicPhrasePlayerService.playMusicPhrase(notes);
```

Comment:
@kevintab95, Should we move the check inside the player service such that ant other services using this player would be safe? (I'm assuming the error is happening because of the player service, right?)

Reply:
Hi @DubeySandeep  -- Yes, the error occurs inside the player service when the MIDI.chordOn method is called. Looks like there is some computation done before executing that (eg. looping through midiSequence) and that's why I thought it should go here. Also, we do not use playMusicPhrase anywhere else currently. Do you think we can leave this as is? Thanks.

Reply:
I think midiSequence is an array (not sure!) and all those computations are basic so I think we don't have to guard them!

>  we do not use playMusicPhrase anywhere else currently. Do you think we can leave this as is?

I would suggest guarding the player services such that it's safe to use in the future!

Reply:
@DubeySandeep, sure I can do that. I noticed the player service has been migrated to Angular. Is there anyway I can use 'AlertsService' there without having to migrate that and make changes in all the places that used AlertsService? Thanks.

Reply:
I think we have an in-review PR for AlertService at #7777. I think we can use it without the migration but not sure!
@bansalnitish or @kevinlee12 can you help @kevintab95 for the above question?

Reply:
@DubeySandeep that should be okay, but we should get that PR in soon.